src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -166,10 +189,17 @@ public void prepareRun(BatchSourceContext context) throws Exception {
 
     // Configure BigQuery input format.
     String temporaryGcsPath = BigQuerySourceUtils.getTemporaryGcsPath(bucket, bucketPath, bucketPath);
-    BigQuerySourceUtils.configureBigQueryInput(configuration,
-                                               DatasetId.of(config.getDatasetProject(), config.getDataset()),
-                                               config.getTable(),
-                                               temporaryGcsPath);
+    try {
+      BigQuerySourceUtils.configureBigQueryInput(configuration,
+        DatasetId.of(config.getDatasetProject(), config.getDataset()),
+        config.getTable(),
+        temporaryGcsPath);
+    } catch (Exception e) {
+      String errorReason = "Failed to configure BigQuery input.";
+      collector.addFailure(String.format("%s %s", errorReason, e.getMessage()), null)
```

Comment:
is it a good idea to set error in error collector here? How will it write this in pipeline error logs.

Reply:
This is taken from the [reference PR](https://github.com/data-integrations/google-cloud/pull/1452)

Ref code:
https://github.com/data-integrations/google-cloud/blob/601f62bd0005255e75bff19db8711a8898358d49/src/main/java/io/cdap/plugin/gcp/gcs/source/GCSSource.java#L115-L118